source/guides/module_guides/bgtm.md

```diff
@@ -58,256 +60,323 @@ In terms of class structure we recommend the following (more detail below):
-
-~~~ruby
-    class ntp (
-      $autoupdate        = $ntp::params::autoupdate,
-      $config            = $ntp::params::config,
-      $config_template   = $ntp::params::config_template,
-      $driftfile         = $ntp::params::driftfile,
-      $keys_enable       = $ntp::params::keys_enable,
-      $keys_file         = $ntp::params::keys_file,
-      $keys_controlkey   = $ntp::params::keys_controlkey,
-      $keys_requestkey   = $ntp::params::keys_requestkey,
-      $keys_trusted      = $ntp::params::keys_trusted,
-      $package_ensure    = $ntp::params::package_ensure,
-      $package_name      = $ntp::params::package_name,
-      $panic             = $ntp::params::panic,
-      $preferred_servers = $ntp::params::preferred_servers,
-      $restrict          = $ntp::params::restrict,
-      $servers           = $ntp::params::servers,
-      $service_enable    = $ntp::params::service_enable,
-      $service_ensure    = $ntp::params::service_ensure,
-      $service_manage    = $ntp::params::service_manage,
-      $service_name      = $ntp::params::service_name,
-      $udlc              = $ntp::params::udlc
-    ) inherits ntp::params {
-~~~
+For instance, the main `ntp` class in the `ntp` module looks like this:
+
+```ruby
+class ntp (
+  Boolean $broadcastclient,
+  Stdlib::Absolutepath $config,
+  Optional[Stdlib::Absolutepath] $config_dir,
+  String $config_file_mode,
+  Optional[String] $config_epp,
+  Optional[String] $config_template,
+  Boolean $disable_auth,
+  Boolean $disable_dhclient,
+  Boolean $disable_kernel,
+  Boolean $disable_monitor,
+  Optional[Array[String]] $fudge,
+  Stdlib::Absolutepath $driftfile,
+ ...
+```
 
 #### `module::install`
 
-The install class must be located in the `install.pp` file, and should contain all of the resources related to getting the software the module manages onto the node.
+The install class must be located in the `install.pp` file. It should contain all of the resources related to getting the software that the module manages onto the node.
 
-The install class must be named `module::install`, as ntp demonstrates:
+The install class must be named `module::install`, as in the `ntp` module:
 
-~~~ ruby
-    class ntp::install inherits ntp {
+``` ruby
+# @api private 
+# This class handles ntp packages. Avoid modifying private classes.
+class ntp::install inherits ntp {
 
-      package { 'ntp':
-        ensure => $package_ensure,
-        name   => $package_name,
-      }
+  if $ntp::package_manage {
 
+    package { $ntp::package_name:
+      ensure => $ntp::package_ensure,
     }
-~~~
+
+  }
+
+}
+```
 
 #### `module::config`
 
 The resources related to configuring the installed software should be placed in a config class. The config class must be named `module::config` and must be located in the `config.pp` file.
 
-See, for example, the ntp module:
+For example, see the `module::config` class in the `ntp` module: TODO: again, do we need this whole class here?
+
```

Comment:
haha, no TODO should never be there... I should have searched for this.

I've cut down the sample now... let me know if I cut anything especially illustrative or anything!